base/docs/Docs.jl

```diff
@@ -244,165 +160,185 @@ is stored as `Union{Tuple{T}, Tuple{T, Any}}`.
+"""
+function fielddoc(binding::Binding, field::Symbol)
     for mod in modules
-        docs = meta(mod)
-        if haskey(docs, T) && isa(docs[T], MultiDoc)
-            fields = docs[T].fields
-            if haskey(fields, k)
-                return fields[k]
+        dict = meta(mod)
+        if haskey(dict, binding)
+            multidoc = dict[binding]
+            if haskey(multidoc.docs, Union{})
+                fields = multidoc.docs[Union{}].data[:fields]
+                haskey(fields, field) && return Markdown.parse(fields[field])
             end
         end
     end
-    fields = join(["`$f`" for f in fieldnames(T)], ", ", ", and ")
+    fields = join(["`$f`" for f in fieldnames(resolve(binding))], ", ", ", and ")
     fields = isempty(fields) ? "no fields" : "fields $fields"
-    Markdown.parse("`$T` has $fields.")
+    Markdown.parse("`$(resolve(binding))` has $fields.")
 end
 
-# Generic Callables
+# As with the additional `doc` methods, this converts an object to a `Binding` first.
+fielddoc(object, field::Symbol) = fielddoc(aliasof(object, typeof(object)), field)
 
-# Modules
+# Object Summaries.
+# =================
 
-function doc(m::Module)
-    md = get_obj_meta(m)
-    md === nothing || return md
+function summarise(binding::Binding, sig)
+    io = IOBuffer()
+    println(io, "No documentation found.\n")
+    if defined(binding)
+        summarise(io, resolve(binding), binding)
+    else
+        println(io, "Binding `", binding, "` does not exist.")
+    end
+    Markdown.parse(seekstart(io))
+end
+
+function summarise(io::IO, λ::Function, binding)
+    kind = startswith(string(binding.var), '@') ? "macro" : "`Function`"
+    println(io, "`", binding, "` is a ", kind, ".")
+    println(io, "```\n", methods(λ), "\n```")
+end
+
+function summarise(io::IO, T::DataType, binding)
+    println(io, "**Summary:**")
+    println(io, "```")
+    println(io,
+        T.abstract ? "abstract" : T.mutable ? "type" : "immutable",
+        " ", T, " <: ", supertype(T)
+    )
+    println(io, "```")
+    if !isempty(fieldnames(T))
+        println(io, "**Fields:**")
+        println(io, "```")
+        pad = maximum(length(string(f)) for f in fieldnames(T))
+        for (f, t) in zip(fieldnames(T), T.types)
+            println(io, rpad(f, pad), " :: ", t)
+        end
+        println(io, "```")
+    end
+    if !isempty(subtypes((T)))
+        println(io, "**Subtypes:**")
+        println(io, "```")
+        for t in subtypes(T)
+            println(io, t)
+        end
+        println(io, "```")
+    end
+end
+
+function summarise(io::IO, m::Module, binding)
     readme = Pkg.dir(string(m), "README.md")
     if isfile(readme)
-        return Markdown.parse_file(readme)
+        println(io, "Displaying the `README.md` for the module instead.\n")
+        println(io, "---\n")
+        println(io, readstring(readme))
+    else
+        println(io, "Additionally, no `README.md` found for module `", m, "`.\n")
```

Comment:
It was in addition to the default no docs message that appears, but it's not really necessary so I've remove the "Additionally".